Report/Chapters/Design.tex

```diff
@@ -8,5 +8,50 @@ \section{Languages and libraries}
 
 \section{High-level architecture}
```

Comment:
Im missing the argumentation as to why we decided to implement our own type system as opposed to using the error messages from GHCi